Budzich Maxim

Zviger opened the pull request.

Zviger committed:
The first version of the application is added.

Zviger committed:
Readme file is added

Zviger committed:
Some fetures are added

The structure of the data received by the RSS parser has been redone.
Improved parser mechanism. Added parameters "limit" and "length".
Added some validation.

Zviger committed:
Added conversion to json format and reworked getting img links

Zviger committed:
Some fetures are added

Added parametr "verbose" and some docstrings.

Zviger committed:
Version of the app is changed

Zviger committed:
Readme file is fixed

Zviger committed:
New version of the app

Project structure is reworked. Setup tool is added.

Zviger committed:
Single quotes replaced

Zviger committed:
Reworked some things

The structure of parsed data has been redone.
Typehinting is added. Etc...

Zviger committed:
Fixed typehinting and etc.

Zviger committed:
Working with time is reworked

Zviger committed:
Added caching and bug fixes

Zviger committed:
Working with docker is added

Zviger committed:
Verion, --help and readme are fixed

Zviger committed:
Docker files are saved

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,21 @@
+with open("requirements.txt") as fp:
+    install_requires = fp.read()
+
+setuptools.setup(
```
попытался сделать как описано в readme:
1) docker-compose up -d 
2) запустить контейнер через exec 
попытался запустить, вот такой аутпут получился 
```bash
# rss_reader https://news.tut.by/rss/geonews/minsk.rss --verbose
2019-11-26 20:25:11,537 - console_app - Program started
2019-11-26 20:25:11,537 - console_app - Parsing https://news.tut.by/rss/geonews/minsk.rss started
2019-11-26 20:25:11,537 - console_app - Connecting with database
2019-11-26 20:25:11,538 - console_app - Load data from database
Traceback (most recent call last):
  File "/usr/local/bin/rss_reader", line 11, in <module>
    load_entry_point('rss-reader==3.1', 'console_scripts', 'rss_reader')()
  File "/usr/local/lib/python3.8/site-packages/rss_reader-3.1-py3.8.egg/app/core.py", line 5, in main
  File "/usr/local/lib/python3.8/site-packages/rss_reader-3.1-py3.8.egg/app/support_files/rss_reader.py", line 53, in exec_console_args
  File "/usr/local/lib/python3.8/site-packages/rss_reader-3.1-py3.8.egg/app/support_files/database.py", line 65, in find_feed_by_link_and_date
  File "/usr/local/lib/python3.8/_strptime.py", line 562, in _strptime_time
    tt = _strptime(data_string, format)[0]
  File "/usr/local/lib/python3.8/_strptime.py", line 317, in _strptime
    raise TypeError(msg.format(index, type(arg)))
TypeError: strptime() argument 0 must be str, not <class 'NoneType'>

```

AlexeiBuzuma replied:
Такая же ситуация, если установить через `pip install .`

Zviger replied:
Забавно, что у меня работает в этих двух случаях прекрасно)

Zviger replied:
Разобрался. Я каким-то образом не закоммитил одну строчку изменения...

AlexeiBuzuma commented on app/support_files/database.py:
```diff
@@ -0,0 +1,95 @@
+from app.support_files.exeptions import FindFeedError, DateError
+
+
+class DB:
```
Я бы назвал этот класс как-нибудь более конкретным именем. 
Все таки это не база данных в прямом понимании. 
Это DBAccessor, DBManager, DBProvider, что угодно, но не просто DB :)

Zviger replied:
Название модуля точно неудачное, а вот по поводу названия класса я бы немного пообщался)

Zviger replied:
Подумал ещё чуть-чуть и понял, что действительно нужно обе вещи рефакторить.

AlexeiBuzuma commented on app/support_files/rss_reader.py:
```diff
@@ -0,0 +1,76 @@
+    exeptions)
+
+
+class Reader:
```
такая же ситуация с названием 
более точно название подошло бы лучше тут. Все таки ридеры могут быть очень-очень разными :)

Zviger replied:
Я полагал, что через название модуля передаётся смысл того, что в нём происходит и решил, что не стоит удлинять название класса.

Zviger committed:
Saving data in html and fb2 are added

Added more logging. Saving images url is reworked.

Zviger committed:
Added color text

Zviger committed:
Tests are added

Zviger committed:
Fixed bag with '--limit' argument

AlexeiBuzuma commented on README.md:
````diff
@@ -0,0 +1,71 @@
+```text
+pip install . --user
+```
+Also, you must have installed and running MongoDB.
````
Распиши пожалуйста подробно, что я должен сделать, чтобы начать пользоваться твоим решением 
идеально написать какой-нибудь баш скрипт, который позволит сделать это все автоматически

Zviger committed:
MongoDB is dockerized

MongoDB now work in Docker container.
Normal working with static files is added
